dpctl/tensor/libtensor/include/kernels/elementwise_functions/expm1.hpp

```diff
@@ -73,6 +74,45 @@ template <typename argT, typename resT> struct Expm1Functor
             const realT x = std::real(in);
             const realT y = std::imag(in);
 
+            // special cases
+            if (std::isinf(x)) {
+                if (x > realT(0)) {
+                    // positive infinity cases
+                    if (!std::isfinite(y)) {
+                        return resT{x, std::numeric_limits<realT>::quiet_NaN()};
+                    }
+                    else if (y == realT(0)) {
+                        return in;
+                    }
+                    else {
+                        return (std::numeric_limits<realT>::infinity() *
+                                    resT{std::cos(y), std::sin(y)} -
+                                realT(1));
+                    }
+                }
+                else {
+                    // negative infinity cases
+                    if (!std::isfinite(y)) {
+                        return resT{-1, 0};
+                    }
+                    else {
+                        return (realT(0) * resT{std::cos(y), std::sin(y)} -
+                                realT(1));
```

Comment:
Perhaps it is better to do:

```suggestion
                        return resT{ realT(-1), std::copysign(realT(0), std::sin(y))};
```